test/evp_test.c

```diff
@@ -1258,6 +1259,13 @@ static int pkey_test_ctrl(EVP_PKEY_CTX *pctx, const char *value)
     if (p != NULL)
         *p++ = 0;
     rv = EVP_PKEY_CTX_ctrl_str(pctx, tmpval, p);
+    if (p != NULL && rv <= 0 && rv != -2) {
```

Comment:
Doesn't this mean we will skip the test if the ctrl fails for some other reason? E.g. typo in evptests? If so that could be bad...i.e. we think we're testing something but actually we're not.